transports/jms/src/test/java/org/mule/compatibility/transport/jms/AbstractBrokerFunctionalTestCase.java

```diff
@@ -20,47 +18,28 @@
 {
 
     @Rule
-    public DynamicPort port = new DynamicPort("port");
-
-    protected BrokerService broker;
-    protected TransportConnector transportConnector;
-    protected String url;
-
+    public ActiveMQBroker amqBroker = new ActiveMQBroker("port");
 
     @Override
     protected void doSetUpBeforeMuleContextCreation() throws Exception
     {
-        url = "tcp://localhost:" + this.port.getValue();
-        startBroker();
+        amqBroker.start();
```

Comment:
I'm not sure what is the benefit of creating a JUnit rule if you have to manually start and stop the broker in an Abstract class.